tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AdoNet/SqlServerDatabaseMonitoringTests.cs

```diff
@@ -0,0 +1,134 @@
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests.AdoNet
+{
+    [Trait("RequiresDockerDependency", "true")]
+    [UsesVerify]
+    public class SqlServerDatabaseMonitoringTests : TracingIntegrationTest
+    {
+        public SqlServerDatabaseMonitoringTests(ITestOutputHelper output)
+            : base("SqlServer", output)
+        {
+            SetServiceVersion("1.0.0");
+            SetEnvironmentVariable("DD_ENV", "testing");
+        }
+
+        public override Result ValidateIntegrationSpan(MockSpan span) => span.IsSqlClient();
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.SystemDataSqlClient), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        public async Task SubmitDbmCommentedSpanspropagationNotSet(string packageVersion)
+        {
+            await SubmitDbmCommentedSpans(packageVersion, string.Empty);
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.SystemDataSqlClient), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        public async Task SubmitDbmCommentedSpanspropagationIntValue(string packageVersion)
+        {
+            await SubmitDbmCommentedSpans(packageVersion, "100");
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.SystemDataSqlClient), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        public async Task SubmitDbmCommentedSpanspropagationRandomValue(string packageVersion)
+        {
+            await SubmitDbmCommentedSpans(packageVersion, "randomValue");
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.SystemDataSqlClient), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        public async Task SubmitDbmCommentedSpanspropagationDisabled(string packageVersion)
+        {
+            await SubmitDbmCommentedSpans(packageVersion, "disabled");
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.SystemDataSqlClient), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        public async Task SubmitDbmCommentedSpanspropagationService(string packageVersion)
+        {
+            await SubmitDbmCommentedSpans(packageVersion, "service");
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.SystemDataSqlClient), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        public async Task SubmitDbmCommentedSpanspropagationFull(string packageVersion)
+        {
+            await SubmitDbmCommentedSpans(packageVersion, "full");
+        }
+
```

Comment:
You _could_ merge all these into a single theory test if you prefer:

```suggestion

        public static IEnumerable<object[]> GetTestData()
            => from packageVersion in PackageVersions.SystemDataSqlClient.SelectMany(x => x).Select(x => (string)x)
               from propagation in new[] { string.Empty, "100", "randomValue", "disabled", "service", "full" }
               select new object[] { packageVersion, propagation };

        [SkippableTheory]
        [MemberData(nameof(GetTestData))]
        [Trait("Category", "EndToEnd")]
```